x/evm/wrappers/bank_test.go

```
@@ -0,0 +1,752 @@
+			evmDenom:    TokenDenom,
+			evmDecimals: 6,
+			amount:      big.NewInt(1e18), // 1 token in 18 decimals
+			recipient:   sdk.AccAddress([]byte("test_address")),
+			expectErr:   "",
+			mockSetup: func(mbk *testutil.MockBankWrapper) {
+				expectedCoin := sdk.NewCoin(TokenDenom, sdkmath.NewInt(1e6)) // 1 token in 6 decimals
+				expectedCoins := sdk.NewCoins(expectedCoin)
+
+				mbk.EXPECT().
+					MintCoins(gomock.Any(), evmtypes.ModuleName, expectedCoins).
+					Return(nil)
+
+				mbk.EXPECT().
+					SendCoinsFromModuleToAccount(
+						gomock.Any(),
+						evmtypes.ModuleName,
+						sdk.AccAddress([]byte("test_address")),
+						expectedCoins,
+					).Return(nil)
+			},
+		},
+		{
+			name:        "success - 18 decimals amount not modified",
+			evmDenom:    TokenDenom,
+			evmDecimals: 18,
+			amount:      big.NewInt(1e18), // 1 token in 18 decimals
+			recipient:   sdk.AccAddress([]byte("test_address")),
+			expectErr:   "",
+			mockSetup: func(mbk *testutil.MockBankWrapper) {
+				expectedCoin := sdk.NewCoin(TokenDenom, sdkmath.NewInt(1e18))
+				expectedCoins := sdk.NewCoins(expectedCoin)
+
+				mbk.EXPECT().
+					MintCoins(gomock.Any(), evmtypes.ModuleName, expectedCoins).
+					Return(nil)
+
+				mbk.EXPECT().
+					SendCoinsFromModuleToAccount(
+						gomock.Any(),
+						evmtypes.ModuleName,
+						sdk.AccAddress([]byte("test_address")),
+						expectedCoins,
+					).Return(nil)
+			},
+		},
+		{
+			name:        "fail - mint coins error",
+			evmDenom:    TokenDenom,
+			evmDecimals: 6,
+			amount:      big.NewInt(1e18),
+			recipient:   sdk.AccAddress([]byte("test_address")),
+			expectErr:   "failed to mint coins to account in bank wrapper",
+			mockSetup: func(mbk *testutil.MockBankWrapper) {
+				expectedCoin := sdk.NewCoin(TokenDenom, sdkmath.NewInt(1e6))
+				expectedCoins := sdk.NewCoins(expectedCoin)
+
+				mbk.EXPECT().
+					MintCoins(gomock.Any(), evmtypes.ModuleName, expectedCoins).
+					Return(errors.New("mint error"))
```

why are the error messages here different? shouldn't the same ones be used?

---

This is an error message that will be returned by the mock suite. It is a simple way to simulate an error when calling the `MintCoins` function of the bank keeper. We don't need to return the real error the bank will return since this approach abstract away the implementation of the function and we just check for a generic error in the wrapper:

```go
	if err := w.BankKeeper.MintCoins(ctx, types.ModuleName, coinsToMint); err != nil {
		return errors.Wrap(err, "failed to mint coins to account in bank wrapper")
	}
```